backend/src/main/java/com/votogether/domain/post/service/PostCommandService.java

```diff
@@ -0,0 +1,358 @@
+    }
+
+    private String updateImage(
+            final String originImageUrl,
+            final String clientUrl,
+            final MultipartFile multipartFile
+    ) {
+        final boolean hasOriginImage = Objects.nonNull(originImageUrl);
+        final boolean hasClientUrl = Objects.nonNull(clientUrl) && !clientUrl.isBlank();
+        final boolean hasNewImage = multipartFile != null && !multipartFile.isEmpty();
+
+        if (hasOriginImage && (!hasClientUrl || hasNewImage)) {
+            imageUploader.delete(originImageUrl);
+        }
+        if (hasNewImage) {
+            return imageUploader.upload(multipartFile);
+        }
+        if (hasOriginImage && hasClientUrl) {
+            return originImageUrl;
+        }
+        return null;
+    }
+
+    private void addNewPostOptions(
+            final Post post,
+            final List<PostOptionUpdateRequest> postOptionUpdates,
+            final int prevPostOptionSize
+    ) {
+        IntStream.range(0, postOptionUpdates.size())
+                .mapToObj(index -> createNewPostOption(index, postOptionUpdates.get(index), prevPostOptionSize))
+                .forEach(post::addPostOption);
+    }
+
+    private PostOption createNewPostOption(
+            final int index,
+            final PostOptionUpdateRequest postOptionUpdate,
+            final int prevPostOptionSize
+    ) {
+        final String imageUrl = imageUploader.upload(postOptionUpdate.getImageFile());
+        return PostOption.builder()
+                .sequence(index + 1 + prevPostOptionSize)
+                .content(postOptionUpdate.getContent())
+                .imageUrl(imageUrl)
+                .build();
+    }
+
+    private void updatePost(final Post post, final PostUpdateRequest postUpdate) {
+        updatePostCategories(post, postUpdate.getCategoryIds());
+        updatePostContentImage(post, postUpdate.getImageUrl(), postUpdate.getImageFile());
+        post.update(postUpdate.getTitle(), postUpdate.getContent(), postUpdate.getDeadline());
+    }
+
+    private void updatePostCategories(final Post post, final List<Long> categoryIds) {
+        post.getPostCategories().clear();
+        postCategoryRepository.flush();
+        addCategories(post, categoryIds);
+    }
+
+    private void updatePostContentImage(
+            final Post post,
+            final String imageUrl,
+            final MultipartFile multipartFile
+    ) {
+        final PostContentImage postContentImage = post.getFirstContentImage();
+        final String currentImageUrl = postContentImage != null ? postContentImage.getImageUrl() : null;
+        final String updatedImageUrl = updateImage(currentImageUrl, imageUrl, multipartFile);
+
+        if (postContentImage == null && updatedImageUrl != null) {
+            post.addContentImage(updatedImageUrl);
+        } else {
+            updateExistingPostContentImage(post, postContentImage, updatedImageUrl);
+        }
```

Comment:
```P2```
```suggestion
        if (postContentImage == null && updatedImageUrl != null) {
            post.addContentImage(updatedImageUrl);
            return;
        }
        
        updateExistingPostContentImage(post, postContentImage, updatedImageUrl);
```

이런식으로 Early Return 방식을 사용하는 것은 어떨까요. 물론 if 중첩이 크게 없어서 가독성 차이가 별로 나진 않지만, 저는 나중을 위해서라도 되도록이면 처음 코드를 작성할 때부터 else 문은 쓰지 않는 것을 권장하는 편입니다.